packages/altive_lints/lib/src/lints/avoid_hardcoded_color.dart

```diff
@@ -48,7 +49,7 @@ class AvoidHardcodedColor extends DartLintRule {
     context.registry.addInstanceCreationExpression((node) {
       final typeName = node.staticType?.getDisplayString();
 
-      if (typeName == 'Color') {
+      if (typeName == 'Color' && !_isInsideColorScheme(node)) {
         reporter.atNode(node, _code);
       }
```

Comment:
![badge](https://img.shields.io/badge/review-imo-important) 

I suggest moving the `_isInsideColorScheme(node)` check to the beginning of this method. By using an early return here, we can avoid redundant checks and improve the code’s readability. 

If `_isInsideColorScheme(node)` returns `true`, we immediately exit, which makes it clear that further processing only applies to nodes outside of a `ColorScheme`. This also eliminates the need to call `_isInsideColorScheme()` multiple times, resulting in a slight performance improvement.